opm/input/eclipse/EclipseState/Grid/LgrCollection.hpp

```diff
@@ -0,0 +1,61 @@
+/*
+ Copyright (C) 2023 Equinor
+  This file is part of the Open Porous Media project (OPM).
+  OPM is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+  OPM is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+  You should have received a copy of the GNU General Public License
+  along with OPM.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef OPM_PARSER_LGR_COLLECTION_HPP
+#define OPM_PARSER_LGR_COLLECTION_HPP
+
+#include <string>
+
+#include <opm/input/eclipse/EclipseState/Util/OrderedMap.hpp>
+#include <opm/input/eclipse/EclipseState/Grid/CarfinManager.hpp>
+#include <opm/input/eclipse/EclipseState/Grid/Carfin.hpp>
+#include <opm/input/eclipse/EclipseState/Grid/EclipseGrid.hpp>
+
+namespace Opm {
+
+    class DeckRecord;
+    class GridDims;
+    class GRIDSection;
+
+
+class LgrCollection {
+public:
+    LgrCollection();
+    LgrCollection(const GRIDSection& gridSection, const EclipseGrid& grid);
+
+   static LgrCollection serializationTestObject();
+
+   explicit LgrCollection(const Deck& deck);
+
+    size_t size() const;
+    bool hasLgr(const std::string& lgrName) const;
+    Carfin& getLgr(const std::string& lgrName);
+    const Carfin& getLgr(const std::string& lgrName) const;
+
+    void addLgr(const EclipseGrid& grid, const DeckRecord& lgrRecord);
+
+    template<class Serializer>
+    void serializeOp(Serializer& serializer)
+    {
+        serializer(m_lgrs);
+    }
+
+private:
+    OrderedMap<Carfin, 8> m_lgrs;
```

Comment:
I think this is a good idea, but the [`CarFin`](https://github.com/OPM/opm-common/blob/ee44c0fbe5c4c6e9fd86c0fca8de7c4ffc63859c/opm/input/eclipse/EclipseState/Grid/Carfin.hpp#L33) type does not meet every requirement of `OrderedMap` in parallel. In particular the element type `T`&mdash;`CarFin` in this instance&mdash;must have a default constructor for the parallel serialisation code to work.  You will also need to add a `serializeOp()` member function to `CarFin`, and an equality operator (`bool operator==(const LgrCollection&) const`) to the `LgrCollection`.

There should probably be a `serializationTestObject()` function too, but that's not an immediate requirement.

Reply:
Thank you for the guidance @bska. I can now compile opm-simulators and pass jenkins tests. Anyway, I am not confident I have added everything needed regarding parallel serialization...will wait for your comments. :)